tests/conftest.py

```diff
@@ -22,11 +24,25 @@ def root_folder():
     return cwd
 
 
-@pytest.fixture()
+@pytest.fixture(autouse=True)
 def data_folder(root_folder):
     yield root_folder / "data"
 
 
 @pytest.fixture()
 def test_folder(root_folder):
     yield root_folder / "tests"
+
+def pytest_generate_tests(metafunc):
+    root = Path().absolute()
+    if "test" in root.parts[-1]:
+        root = root.parent
+
+    params = []
+    for step in ['raw', 'interim']:
+        for filename in listdir(join(root, 'data', step)):
+            if '.tsv' in filename:
+                params.append((step, filename[:-4]))
```

Comment:
Some concerns on robustness here.

* Using `in` scans complete filename (including all leading directories), might accidentally include search string.
 * No dependence on magic number 4

Code could be improved using pathlib.

```py
params = []
for step in ['raw', 'interim']:
    for file in root.joinpath('data', step).iterdir():
        if file.suffix == '.tsv':
            params.append((step, filename.with_suffix('')
```